### PR TITLE
ci: update build to use TypeScript 2.3.2

### PR DESCRIPTION
**Pull request by chuckjaz**
**Please check if the PR fulfills these requirements**
**What kind of change does this PR introduce?** (check one with "x")
```
[x] CI related changes
```

**What is the current behavior?** (You can also link to an open issue here)

Angular uses TypeScript 2.1.5 to build.

**What is the new behavior?**

Angular uses TypeScript 2.3.2 to build.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: integration/typings_test_ts23/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "angular-integration",
+  "description": "Assert that users with TypeScript 2.2 can type-check an Angular application",
```
Comment (alexeagle) [inline integration/typings_test_ts23/package.json]: update 2.2 -> 2.3

File: integration/typings_test_ts23/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "target": "es5",
+    "lib": [
+      "es5",
+      "dom",
+      "es2015.collection",
+      "es2015.iterable",
+      "es2015.promise"
+    ],
+    "types": [],
+    "strictNullChecks": true
```
Comment (alexeagle) [inline integration/typings_test_ts23/tsconfig.json]: Ideally we should support the new "strict": true setting.
I dunno if that will be green though.

File: packages/http/test/backends/xhr_backend_spec.ts [hunk 3]
```diff
@@ -334,7 +334,7 @@ export function main() {
         it('should use FormData body and detect content type header to the request', () => {
           const body = new FormData();
           body.append('test1', 'val1');
-          body.append('test2', 123456);
+          body.append('test2', '' + 123456);
```
Comment (alexeagle) [inline packages/http/test/backends/xhr_backend_spec.ts]: 'val2' maybe?

**Review by alexeagle — APPROVED** [review]

File: integration/typings_test_ts23/tsconfig.json (hunk 2 above)
Comment (chuckjaz) [inline integration/typings_test_ts23/tsconfig.json]: I will change it to "strict" and fix any errors I find. I will back off if I have to change too much.

File: packages/http/test/backends/xhr_backend_spec.ts (hunk 3 above)
Comment (chuckjaz) [inline packages/http/test/backends/xhr_backend_spec.ts]: `'123456'` instead?

File: packages/compiler/src/i18n/extractor_merger.ts [hunk 4]
```diff
@@ -456,7 +456,7 @@ class _Visitor implements html.Visitor {
         0);
 
     if (significantChildren == 1) {
-      for (let i = this._messages.length - 1; i >= startIndex; i--) {
+      for (let i = this._messages.length - 1; i >= startIndex !; i--) {
```
Comment (IgorMinar) [inline packages/compiler/src/i18n/extractor_merger.ts]: what does a trailing bang mean? cast to not null?

**Review by IgorMinar — APPROVED** [review]
lgtm otherwise

File: packages/compiler/src/i18n/extractor_merger.ts (hunk 4 above)
Comment (Toxicable) [inline packages/compiler/src/i18n/extractor_merger.ts]: Yup, Non-null assertion operator, basically means `startIndex` is neither `null` or `undefined`
https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#non-null-assertion-operator

Comment (awerlang) [inline packages/compiler/src/i18n/extractor_merger.ts]: clang-format isn't formatting it properly, there should be no space before it. Not sure if it's just configuration though.

**Comment by alexeagle** [comment]
We are just on an old version of clang-format. Updating it is disruptive
since we would re-format everything, and not very beneficial.

**Pull request merged**